packages/webcomponents/src/components/business-form-stepped/business-core-info/business-core-info-form-step.tsx

```diff
@@ -23,17 +24,81 @@ import { CoreBusinessInfo, ICoreBusinessInfo } from '../../../api/Business';
   styleUrl: 'business-core-info-form-step.scss',
 })
 export class BusinessCoreInfoFormStep {
-  @Prop() formController: FormController;
+  @Prop() authToken: string;
+  @Prop() businessId: string;
+  @State() formController: FormController;
   @State() errors: any = {};
-  @State() genericInfo: ICoreBusinessInfo = {};
+  @State() coreInfo: ICoreBusinessInfo = {};
+  @Event({ bubbles: true }) submitted: EventEmitter<{ data?: any}>;
+  @Event() formLoading: EventEmitter<boolean>;
+  @Event() serverError: EventEmitter<{ data: any, message: BusinessFormServerErrors }>;
 
   constructor() {
     this.inputHandler = this.inputHandler.bind(this);
+    this.sendData = this.sendData.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  private api: any;
+
+  get businessEndpoint() {
+    return `entities/business/${this.businessId}`
+  }
+
+  private async fetchData() {
+    this.formLoading.emit(true);
+    try {
+      const response: IApiResponse<IBusiness> = await this.api.get(this.businessEndpoint);
+      this.coreInfo = new CoreBusinessInfo(response.data);
+      this.formController.setInitialValues({ ...this.coreInfo });
+    } catch (error) {
+      this.serverError.emit({ data: error, message: BusinessFormServerErrors.fetchData });
+    } finally {
+      this.formLoading.emit(false);
+    }
+  }
+
+  private async sendData(onSuccess?: () => void) {
+    this.formLoading.emit(true);
+    try {
+      const payload = parseCoreInfo(flattenNestedObject(this.formController.values.getValue()));
+      const response = await this.api.patch(this.businessEndpoint, JSON.stringify(payload));
+      this.handleResponse(response, onSuccess);
+    } catch (error) {
+      this.serverError.emit({ data: error, message: BusinessFormServerErrors.patchData });
+    } finally {
+      this.formLoading.emit(false);
+    }
+  }
+  
+  handleResponse(response, onSuccess) {
+    if (response.error) {
+      this.serverError.emit({ data: response.error, message: BusinessFormServerErrors.patchData });
+    } else {
+      onSuccess();
+    }
+    this.submitted.emit({ data: response });
+  }
+
+  @Method()
+  async validateAndSubmit(onSuccess: () => Promise<void>) {
```

Comment:
This method doesn't need to be async because is not returning the promise:

Could be like this
```
@Method()
  validateAndSubmit({ onSuccess }) {
    this.formController.validateAndSubmit(() => this.sendData(onSuccess));
  };
```